### DISCUSSION TITLE
supabase.auth.getSession() is potentially insecure with Sveltekit

### DISCUSSION THREAD

**Discussion by julien-blanchon** [supabase/supabase#22353] [Questions] [answered]
Since the last update I have the following message that keep popping up in the server console:

`Using supabase.auth.getSession() is potentially insecure as it loads data directly from the storage medium (typically cookies) which may not be authentic. Prefer using supabase.auth.getUser() instead. To suppress this warning call supabase.auth.getUser() before you call supabase.auth.getSession().`

I was initially using `auth-helpers-sveltekit` but I've switched to `supabase/ssr` since it's seems to be the new standard. However the message is still here, and I'm using pretty much the same code as the documentation. I don't find anything about `auth.getUser()` related to `auth.getSession()` in the documentation

**Comment by julien-blanchon** [comment]
Maybe: 

```typescript
event.locals.getSession = async () => {
		const {
			data: { session }
		} = await event.locals.supabase.auth.getSession();
		const { data: {
			user
		} } = await event.locals.supabase.auth.getUser()
		if (user?.id !== session?.user.id) {
			await event.locals.supabase.auth.signOut()
			return null
		}
		return session;
	};
```

calling `auth.getUser()` before effectivelly stop the warning. But I'm not sure what kind of check I must do

**Comment by charislam** [comment, marked as answer]
Thanks for bringing this up, @julien-blanchon!

We've changed our thinking on what the best default is to recommend, when trying to balance between:
- Security of not letting devs think that an unchecked `getSession` response is safe
- Performance of not unnecessarily contacting the Auth server

Sorry if it's been a bit confusing! We've updated our docs to what we now recommend, which you can see here: https://supabase.com/docs/guides/auth/server-side/creating-a-client?framework=sveltekit

Basically, you should:
- Call `getUser`
- Check for whether `getUser` errored, which means the user's JWT signature is invalid. Then you should abort by returning a null session

I think your way does this as well (different paths to the same thing!)

Hope that makes sense!

P.S. I do recommend returning the user as well if that code is in a server hook, so you don't have to call `getUser` again if you need it on a page.

**Comment by hmnd** [comment]
@charislam I understand the intention with this warning, but it's impossible to silence when I know I'm properly verifying with getUser. In my case, I need to use it with `createSupabaseLoadClient` from `auth-helpers`, and because it serializes the session as JSON, I get a warning every time I call that function or eg. spread the session into a new object, etc.

At the very least, could the warning only be displayed once per session or could we please have an option to disable it? I see that now it's gated behind a non-existent `__suppressUserWarning` key on session.

For reference the warning I'm getting is below. It's a different one from the one this discussion originally addresses:
> Using the user object as returned from supabase.auth.getSession() or from some supabase.auth.onAuthStateChange() events could be insecure! This value comes directly from the storage medium (usually cookies on the server) and many not be authentic. Use supabase.auth.getUser() instead which authenticates the data by contacting the Supabase Auth server.

**Reply by charislam** [reply]
We've been getting some requests for improvements to this. Do you mind adding your thoughts to [the main issue](https://github.com/supabase/auth-helpers/issues/755)? It will help our team track all the feedback in one place 🙏🏼

**Comment by julien-blanchon** [comment]
Seriously this is completly destroying supabase, I'm thinking switching to another solution
